docs/design/CONFIG_MANAGEMENT.md

```diff
@@ -0,0 +1,238 @@
+Config Management Design
+===================
+
+## Motivation
+
+[Issue 43][issue_43] specifies the need for lifecycle management of Envoy Gateway system components that run as Go
+routines.
+
+## Goals
+
+* Gracefully manage the services that comprise Envoy Gateway.
+
+## Non-Goals
+
+* Provide a detailed Envoy Gateway configuration specification.
+
+## Proposal
+
+Introduce a `serve` argument to the `envoy-gateway` command. The `serve` argument encapsulates the management of Envoy
+Gateway.
+```go
+// gateway/internal/cmd/root.go
+
+package cmd
+
+import (
+	"github.com/spf13/cobra"
+
+	"github.com/envoyproxy/gateway/internal/cmd/serve"
+)
+
+// GetRootCommand returns the root cobra command to be executed
+// by main.
+func GetRootCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "envoy-gateway",
+		Short: "Manages Envoy Proxy as a standalone or Kubernetes-based application gateway",
+	}
+
+	cmd.AddCommand(serve.AddCommand())
+
+	return cmd
+}
+```
+
+The serve package provides the following functionality:
+- Allows the serve package to be run as a Cobra Command.
+- Validates, processes, etc. command line flags and the Envoy Gateway config file. 
+- Constructs a object, e.g. `GatewayServer`, based on the validated flags/config. This object encapsulates all the
+necessary parameters for running Envoy Gateway.
+- Runs Envoy Gateway based on the provided `GatewayServer`.
+```go
+// gateway/internal/cmd/serve/serve.go
+
+package serve
+
+import (
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"github.com/envoyproxy/gateway/apis/config/v1alpha1"
+)
+
+// AddCommand allows the serve package to be run as a Cobra Command.
+func AddCommand() *cobra.Command {
+	flags := NewGatewayFlags()
+
+	gatewayCfg, _ := NewGatewayConfig()
+
+	cmd := &cobra.Command{
+		Use:   "serve",
+		Short: "Serve serves Envoy Gateway",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// If provided, validate flags and config file.
+			// If provided, load the envoy gateway config file.
+			// Construct a GatewayServer object from the GatewayConfig type.
+			// Run Envoy Gateway using the provided GatewayServer.
+		},
+	}
+
+	return cmd
+}
+
+// GatewayFlags contains Envoy Gateway configuration flags.
+type GatewayFlags struct {
+	// Add fields for command line flags, e.g. config file, enabled services, etc.
+}
+
+// NewGatewayFlags will create a new GatewayFlags with default values.
+func NewGatewayFlags() *GatewayFlags {
+	return &GatewayFlags{
+		// Set flag defaults, e.g. enabled services, config file path, etc.
+	}
+}
+
+// ValidateGatewayFlags validates Envoy Gateway's configuration flags.
+func ValidateGatewayFlags(f *GatewayFlags) error {
+}
+
+// NewGatewayConfig will create a new GatewayConfig with default values.
+func NewGatewayConfig() (*v1alpha1.GatewayConfig, error) {
```

Comment:
* I believe we had [decided](https://github.com/envoyproxy/gateway/pull/16/files#r861377020) to name this Bootstrap/BootstrapConfig 
* `NewGatewayConfig` generates a default config so if `envoy-gateway serve` is called with args (`--config-file <boostrap-file>` or `--config-yaml <boostrap-yaml>`), will the config from `NewGatewayConfig` be discarded or merged with the config passed by the user ?

Reply:
nvm you've answered this below - its default config or user specified config

Reply:
> I believe we had [decided](https://github.com/envoyproxy/gateway/pull/16/files#r861377020) to name this Bootstrap/BootstrapConfig

@arkodg [commit aee4939](https://github.com/envoyproxy/gateway/commit/aee49391d8b46e28716b715e10fbc86709622369) resolves this issue.

Reply:
I like the idea of having a default config that the user doesn't need to explicitly learn before starting the process. This is in line with the project's [goals](https://github.com/envoyproxy/gateway/blob/main/GOALS.md#batteries-included). 
I do see cases where `v1/envoy-gateway` works on platform X but `v2/envoy-gateway` doesn't. We
can handle such cases using a subcommand - `v1/envoy-gateway gen-default-config > v1_default_config.yaml` which can then be explicitly used in `v2/envoy-gateway -c v1_default_config.yaml` and ensure smooth upgrades (since the bootstrap API will be backwards compatible)

Reply:
> I like the idea of having a default config that the user doesn't need to explicitly learn before starting the process.

IMO, an even better solution is for EG to run without requiring a config. For example:
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: envoy-gateway-example
  namespace: envoy-gateway-system
spec:
  template:
    spec:
      containers:
      - args:
        - serve
        command: ["envoy-gateway"]
...
```

In this example, envoy-gateway `serve` will expect to run in a Kubernetes cluster, using default Kubernetes configuration parameters.

Thoughts @youngnick @LukeShu?